src/promise.js

```diff
@@ -35,3 +35,36 @@ export function pause (seconds) {
     setTimeout(resolve, seconds * 1000)
   })
 }
+
+/**
+ * Polls for the presence of a value fetched by a function, within a deadline.
+ * @param {Object} params
+ * @param {number} params.interval Interval between fetchValue checks, in secondsgit
+ * @param {number} params.deadline Deadline, in seconds
+ * @param {function} fetchValue
+ * @returns {Promise}
```

Comment:
We should use a template parameter `T`: 
callback returns `T` or `undefined`
poll returns `Promise<T>`

see https://stackoverflow.com/questions/16017627/document-generic-type-parameters-in-jsdoc